chainstate/tx-verifier/src/transaction_verifier/input_output_policy/tests/homomorphism.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{
+    chain::{
+        config::ChainType, output_value::OutputValue, stakelock::StakePoolData,
+        timelock::OutputTimeLock, AccountNonce, AccountSpending, Destination, NetUpgrades, PoolId,
+        TxOutput,
+    },
+    primitives::{per_thousand::PerThousand, Amount, H256},
+};
+use crypto::{
+    random::{CryptoRng, Rng},
+    vrf::{VRFKeyKind, VRFPrivateKey},
+};
+use rstest::rstest;
+use test_utils::{
+    random::{make_seedable_rng, Seed},
+    split_value,
+};
+
+use super::*;
+
+fn create_stake_pool_data(rng: &mut (impl Rng + CryptoRng), atoms_to_stake: u128) -> StakePoolData {
+    let (_, vrf_pub_key) = VRFPrivateKey::new_from_rng(rng, VRFKeyKind::Schnorrkel);
+    StakePoolData::new(
+        Amount::from_atoms(atoms_to_stake),
+        Destination::AnyoneCanSpend,
+        vrf_pub_key,
+        Destination::AnyoneCanSpend,
+        PerThousand::new(0).unwrap(),
+        Amount::ZERO,
+    )
+}
+
+// Create 2 transactions. The first decommissions the pool and transfer some amount.
+// The second one spends share from delegation and also transfers some amount.
+// Actual amounts and the number of inputs/outputs used is random.
+// Process both transactions using a single ConstrainedValueAccumulator.
+// Then process them again but using separate accumulators per tx.
+// The resulting objects must be equal.
```

Comment:
So this test is OK, but I have a few comments:

1. Put more emphasis on Case A and Case B, and then their combination Case C = A + B. I would explicitly add in the comment here that this test proves that f(a+b)=f(a)+f(b).
2. This is a very confined case. I would randomize this test much more than whatever is provided here, but I don't know the difficulty of doing this and whether this is urgent. You can probably do this later once you're done with the important stuff. Add a note about it. If it's gonna take you a week, then forget about it.
3. The `fee1` and `fee2` variables better be called after the convention, `fee_a` and `fee_b`, and then show that `fee_a`+`fee_b`=`fee_c`.